### PR TITLE
[CN-499] Add operator support for restoring from local backups

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Added local backup list endpoint
- Updated external backup upload logic to only upload the latest single member backup
- Added restore local agent

### CODE REVIEW

File: backup.go [hunk 1]
```diff
@@ -100,17 +108,76 @@ func (p *backupCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
...
+
+		if len(backupUUIDs) != 1 && len(backupUUIDs) <= req.MemberID {
+			httpError(w, http.StatusBadRequest)
+			return
+		}
+
+		// If there is only one backup, members are isolated. No need for memberID
+		if len(backupUUIDs) == 1 {
+			req.MemberID = 0
+		}
```
Comment (cagric0) [inline backup.go:161]: This code snippet can be extracted to a shared function. There are some other places where we have same logic.

File: restore_local.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		return subcommands.ExitFailure
+	}
+
+	id, err := parseID(r.Hostname)
+	if err != nil {
+		log.Println("Parse error", err.Error())
+		return subcommands.ExitFailure
+	}
+
+	lock := filepath.Join(r.BackupBaseDir, restoreLocalLock+r.RestoreID)
```
Comment (cagric0) [inline restore_local.go:67]: We should find a way to delete old restore locks when new restore is started.

**Review by cagric0 — COMMENTED** [review]
Good improvement! I left a few comments and also please do not forget to address linter errors!

File: restore_local.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+	if len(backupUUIDs) != len(destBackupUUIDS) {
+		return fmt.Errorf("Local backup count %d and hot-restart folder %d count are not equal", len(backupUUIDs), len(destBackupUUIDS))
+	}
+
+	if backupUUIDs[id].Name() != destBackupUUIDS[id].Name() {
+		log.Printf("Hot-restart folder name %s and backup UUID folder are not the same %s", destBackupUUIDS[id].Name(), backupUUIDs[id].Name())
+	}
+
+	backup := backupUUIDs[id].Name()
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply restore_local.go:116]: Sure, will do

File: backup.go [hunk 4]
```diff
@@ -100,17 +108,73 @@ func (p *backupCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
 	return subcommands.ExitSuccess
 }
 
+type backupService struct{}
```
Comment (SeriyBg) [inline backup.go:111]: Maybe I'm missing some point, but it seems that this struct is redundant, as we can pass the function as a value directly to the `router.HandleFunc`, so for me the struct seems to no serve any purpose

**Review by SeriyBg — APPROVED** [review]
Left one comment, but other than that LGTM

File: backup.go (hunk 4 above)
Comment (mtyazici) [reply backup.go:111]: I agree it does not have a purpose, my intention was to emphasize that the backup handler belonged to a different service. If you don't think we need to do that, I can just use a function without a receiver.

Comment (SeriyBg) [reply backup.go:111]: I think less code is always better than more code 😄 
For the `uploadService`, there is a state, so the struct makes sense there. But here I would removing it and use just the `func`

Comment (mtyazici) [reply backup.go]: Alright, I deleted the struct :+1:

**Review by cagric0 — APPROVED** [review]
LGTM!

**Pull request merged**